common4j/src/main/com/microsoft/identity/common/java/cache/SharedPreferencesAccountCredentialCacheWithMemoryCache.java

```diff
@@ -90,6 +86,7 @@ private void load() {
                 Logger.info(methodTag, "Loaded " + mCachedAccountRecordsWithKeys.size() + " AccountRecords");
                 mCachedCredentialsWithKeys = loadCredentialsWithKeys();
                 Logger.info(methodTag, "Loaded " + mCachedCredentialsWithKeys.size() + " Credentials");
+                new Thread(() -> removeSha1ApplicationIdentifierAccessTokensIfNeeded()).start();
```

Comment:
>new Thread(() -> removeSha1ApplicationIdentifierAccessTokensIfNeeded()).start(); [](http://example.com/codeflow?start=16&length=80)

Since this is already its own thread, there's no need to take the perf penalty of spinning up another single-use thread.  We can just call this method here.

I would probably take the Thread function on line 77 and instead turn that into a method called 'initialize()' that does load() and then removeSha1*().  That should have a general catch around it so we don't risk remove throwing and crashing us with an uncaught top level exception.

I suggest doing it that way because the load() isn't dependent on the cleanup to finish given the requirements you've set forth, and we can let other apps come in and make reads even while we're doing this work of iterating and removing.

Reply:
Addressed here: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/2058/commits/3e147324a108f039db78dfbdbbc2581f87b44919
Thanks Nick!